junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,544 @@
+        val methods = jUnitAnnotatedMethods(clSym)
+        val mlMethods = mlSymOption.map(jUnitAnnotatedMethods)
+
+        def liftAnnotations(methodSymbol: Symbol): List[Tree] = {
+          val annotations = methodSymbol.annotations
+          def clone(t: Tree): Tree = {
+            t match {
+              case lit: Literal => lit
+
+              case Select(t2, name) => Select(clone(t2), name)
+
+              case ValDef(mod, name, tpt, rhs) =>
+                ValDef(mod, name, tpt, clone(rhs))
+
+              case _ => t
+            }
+          }
+          def liftList(params: List[Tree]): List[Tree] =
+            params.map(clone)
+
+          // Find ad report unsupported JUnit annotations
+          annotations.foreach {
+            case ann if ann.atp.typeSymbol == testAnnotationSymbol && ann.original.isInstanceOf[Block] =>
+              reporter.error(ann.pos, "@Test(timeout = ...) is not " +
+                "supported in Scala.js JUnit Framework")
+
+            case ann if ann.atp.typeSymbol == fixMethodOrderAnnotationSymbol =>
+              reporter.error(ann.pos, "@FixMethodOrder(...) is not supported " +
+                "in Scala.js JUnit Framework")
+
+            case _ => // all is well
+          }
+
+          // Collect lifted representations of the JUnit annotations
+          annotations.collect {
+            case ann if annotationWhiteList.contains(ann.tpe.typeSymbol) =>
+              val args = if (ann.args != null) liftList(ann.args) else Nil
+              mkNewInstance(TypeTree(ann.tpe), args)
+          }
+        }
+
+        def defaultMethodMetadata(tpe: TypeTree)(mtdSym: MethodSymbol): Tree = {
+          val annotations = liftAnnotations(mtdSym)
+          mkNewInstance(tpe, List(
+              Literal(Constant(mtdSym.nameString)),
+              mkList(annotations)))
+        }
+
+        def mkList(elems: List[Tree]): Tree = {
+          val array = ArrayValue(TypeTree(definitions.ObjectTpe), elems)
+          val castArray = gen.mkCast(array, typeOf[Array[Object]])
+          val wrappedArray = gen.mkMethodCall(
+              definitions.PredefModule,
+              definitions.wrapArrayMethodName(definitions.ObjectTpe),
+              Nil, List(castArray))
+          gen.mkMethodCall(definitions.List_apply, List(wrappedArray))
+        }
+
+        def mkMethodList(tpe: TypeTree)(testMethods: List[MethodSymbol]): Tree =
+          mkList(testMethods.map(defaultMethodMetadata(tpe)))
+
+        val getJUnitMethodRhs = {
+          mkNewInstance(
+              TypeTree(jUnitClassMetadataType),
+              List(
+                mkList(liftAnnotations(clSym)),
+                gen.mkNil,
+                mkMethodList(jUnitMethodMetadataTypeTree)(methods),
+                mlMethods.fold(gen.mkNil)(mkMethodList(jUnitMethodMetadataTypeTree))
+          ))
+        }
+
+        val local = methods.head
+        val getJUnitMetadataSym = local.cloneSymbol
```

Comment:
Why not use `classSym.newMethod(newTermName("metadata"))` like you did for `invoke`?